packages/explorer/src/common.ts

```diff
@@ -1,22 +1,36 @@
 import { Chain, anvil, garnet, redstone } from "viem/chains";
 
-export const chains: Partial<Record<number, Chain>> = {
+const supportedChains = [anvil, garnet, redstone];
+
+export type SupportedChainIds = (typeof supportedChains)[number]["id"];
+export type SupportedChainNames = "anvil" | "garnet" | "redstone";
+
+export const chainIdName: Record<SupportedChainIds, SupportedChainNames> = {
+  [anvil.id]: "anvil",
+  [garnet.id]: "garnet",
+  [redstone.id]: "redstone",
+} as const;
+
+export const chainNameId: Record<SupportedChainNames, SupportedChainIds> = {
+  anvil: anvil.id,
+  garnet: garnet.id,
+  redstone: redstone.id,
+};
+
+export const chains: Record<SupportedChainIds, Chain> = {
   [anvil.id]: anvil,
-  [redstone.id]: redstone,
   [garnet.id]: garnet,
+  [redstone.id]: redstone,
 };
 
-export function getChain() {
-  const chainId = Number(process.env.NEXT_PUBLIC_CHAIN_ID || anvil.id);
-  const chain = chains[chainId];
-  if (!chain) {
-    throw new Error(`Chain ID ${chainId} not supported. Supported chains are: ${Object.keys(chains).join(", ")}.`);
+export function isValidChainId(chainId: number): asserts chainId is SupportedChainIds {
+  if (!(chainId in chainIdName)) {
+    throw new Error(`Invalid chain id. Supported chains are: ${Object.keys(chainIdName).join(", ")}.`);
   }
-
-  return chain;
 }
 
-export function isAnvil() {
-  const chain = getChain();
-  return chain.id === anvil.id;
+export function isValidChainName(name: string | string[] | undefined): asserts name is SupportedChainNames {
```

Comment:
```suggestion
export function validateChainName(name: string | string[] | undefined): asserts name is SupportedChainNames {
```

something named `isValid` makes me think this should return a boolean, not throw

Reply:
ah very good point, will do